core/src/SYCL/Kokkos_SYCL_Instance.cpp

```diff
@@ -273,11 +273,9 @@ sycl::device_ptr<void> SYCLInternal::scratch_flags(const std::size_t size) {
 
     m_scratchFlags = reinterpret_cast<size_type*>(r->data());
   }
-  m_queue->memset(m_scratchFlags, 0, m_scratchFlagsCount * sizeScratchGrain);
-  fence(*m_queue,
-        "Kokkos::Experimental::SYCLInternal::scratch_flags fence after "
-        "initializing m_scratchFlags",
-        m_instance_id);
+  auto memset_event = m_queue->memset(m_scratchFlags, 0,
+                                      m_scratchFlagsCount * sizeScratchGrain);
+  m_queue->ext_oneapi_submit_barrier(std::vector{memset_event});
```

Comment:
This change seems unrelated.
Also why `sycl::queue::ext_oneapi_submit_barrier()` instead of regular `sycl::queue::submit_barrier`?

Reply:
I need this change for the tests checking fences. 
`sycl::queue::submit_barrier` doesn't exist (anymore).

Reply:
> I need this change for the tests checking fences.

I do not understand which tests you are referring to.

Reply:
https://github.com/kokkos/kokkos/blob/develop/core/unit_test/tools/TestEventCorrectness.hpp#L391-L473